presto-main/src/main/java/io/prestosql/operator/aggregation/AggregationFromAnnotationsParser.java

```diff
@@ -70,89 +70,104 @@ public static ParametricAggregation parseFunctionDefinitionWithTypesConstraint(C
+            String name,
+            AggregationHeader header,
+            Class<?> stateClass,
+            List<AggregationImplementation> exactImplementations,
+            List<AggregationImplementation> nonExactImplementations)
     {
-        ParametricImplementationsGroup.Builder<AggregationImplementation> implementationsBuilder = ParametricImplementationsGroup.builder();
-        AggregationHeader header = parseHeader(aggregationDefinition);
-        boolean deprecated = aggregationDefinition.getAnnotationsByType(Deprecated.class).length > 0;
-
-        for (Class<?> stateClass : getStateClasses(aggregationDefinition)) {
-            Method combineFunction = getCombineFunction(aggregationDefinition, stateClass);
-            Method outputFunction = getOnlyElement(getOutputFunctions(aggregationDefinition, stateClass));
-            for (Method inputFunction : getInputFunctions(aggregationDefinition, stateClass)) {
-                Optional<Method> removeInputFunction = getRemoveInputFunction(aggregationDefinition, inputFunction);
-                AggregationImplementation implementation = parseImplementation(aggregationDefinition, header, stateClass, inputFunction, removeInputFunction, outputFunction, combineFunction);
-                implementationsBuilder.addImplementation(implementation);
-            }
+        ImmutableList.Builder<ParametricAggregation> functions = ImmutableList.builder();
+
+        // create a separate function for each exact implementation
+        for (AggregationImplementation exactImplementation : exactImplementations) {
+            functions.add(new ParametricAggregation(
+                    signatureWithName(name, exactImplementation.getSignature()),
+                    header,
+                    stateClass,
+                    ParametricImplementationsGroup.of(exactImplementation).withAlias(name)));
         }
+        // if there are non-exact functions, create a generic function
+        if (!nonExactImplementations.isEmpty()) {
+            ParametricImplementationsGroup.Builder<AggregationImplementation> implementationsBuilder = ParametricImplementationsGroup.builder();
+            nonExactImplementations.forEach(implementationsBuilder::addImplementation);
+            ParametricImplementationsGroup<AggregationImplementation> implementations = implementationsBuilder.build();
+            functions.add(new ParametricAggregation(
+                    signatureWithName(name, implementations.getSignature()),
+                    header,
+                    stateClass,
+                    implementations.withAlias(name)));
+        }
+        return functions.build();
+    }
 
-        ParametricImplementationsGroup<AggregationImplementation> implementations = implementationsBuilder.build();
-        return new ParametricAggregation(implementations.getSignature(), header, implementations, deprecated);
+    private static boolean isExact(Signature signature)
```

Comment:
I'm not sure I understand what notion this method is trying to capture (and hence, whether `isExact` is a good name for it)

Reply:
The annotated function framework handles exact functions (by this definition) differently than generic functions, so we have to split them.